NewForce-at-Mountwest/bangazon-api-the-thunder-lobsters
# Dam payment type

# Description
Added functionality to GET, POST, PUT, and DELETE Products from the Bangazon API. Added a test suite for integration testing of this functionality.



## Related Ticket(s)
https://trello.com/c/JDKXk1MH/3-ticket-3-allow-developers-to-access-payment-type-resource

## Steps to Test
* Pull down these changes on a new branch
* Open a new SQL query for your database
* Copy the following code into the new SQL query and execute it:
INSERT INTO PaymentType (AcctNumber, Name, CustomerId)
VALUES ('XO1X36', 'Bangazon Plus', '2')
INSERT INTO PaymentType (AcctNumber, Name, CustomerId)
VALUES ('XO11010', 'Bangazon Gold Elite Plus Ultimate', '3')
INSERT INTO Customer (FirstName, LastName, CreationDate, LastActiveDate)
VALUES ('Json', 'Doe', '2020-12-01 03:03:03', '2020-12-01 03:03:03')
INSERT INTO Customer (FirstName, LastName, CreationDate, LastActiveDate)
VALUES ('Mke', 'Hunt', '2020-12-01 03:03:03', '2020-12-01 03:03:03')

* Edit AppSettings.json (in BangazonAPI) and DatabaseFixture.cs (in TestBangazonAPI) to reference the SQL server running on your machine, rather than SQLEXPRESS03.
* Press CTRL + F5 to start the API
Change the URL of your browser to end in api/PaymentType
* Copy and paste this URL into Postman
* Using Postman, ensure that you can GET a list of all products from this URL.
* Ensure that you can add an ID to the end of the URL to GET a single product resource
* Copy JSON from your GET result and use it to POST a new product
* Use PUT to edit the product you just POSTed
* Use GET to ensure this product was edited
* DELETE the product you created
* Use GET to make sure the product is now removed
* In Visual Studio, navigate to the Test Explorer and run all tests. Ensure that all 5 tests succeed.


# Checklist:
